bin/pre-commit-hook.js

```diff
@@ -3,28 +3,78 @@
 const execSync = require( 'child_process' ).execSync;
 const spawnSync = require( 'child_process' ).spawnSync;
 const chalk = require( 'chalk' );
+const fs = require( 'fs' );
+const prettier = require( 'prettier' );
+const path = require( 'path' );
 
-console.log( '\nBy contributing to this project, you license the materials you contribute ' +
+console.log(
+	'\nBy contributing to this project, you license the materials you contribute ' +
 	'under the GNU General Public License v2 (or later). All materials must have ' +
-	'GPLv2 compatible licenses — see .github/CONTRIBUTING.md for details.\n\n' );
+	'GPLv2 compatible licenses — see .github/CONTRIBUTING.md for details.\n\n'
+);
 
 // Make quick pass over config files on every change
 require( '../server/config/validate-config-keys' );
 
 const files = execSync( 'git diff --cached --name-only' )
 	.toString()
 	.split( '\n' )
-	.map( ( name ) => name.trim() )
-	.filter( ( name ) => name.endsWith( '.js' ) || name.endsWith( '.jsx' ) );
+	.map( name => name.trim() )
+	.filter( name => name.endsWith( '.js' ) || name.endsWith( '.jsx' ) );
 
+/**
+ * Returns true if the given text contains '@ format' (without the space after @).
+ * within its first docblock. False otherwise.
+ *
+ * Note: I only don't write out the actual @ f o r m a t, because then this
+ * file would also get formatted!
+ *
+ * @param {*} text text to scan for the format keyword within the first docblock
+ */
+const shouldFormat = text => {
+	const docBlockStart = '/**';
+	const docBlockEnd = '*/';
+
+	const firstDocBlockStartIndex = text.indexOf( docBlockStart );
```

Comment:
I don't think that extracting these constants helps clarify the logic at all. I think being able to see in one line what we are doing here makes it read easier. everyone should recognize what `test.indexOf( '/**' )` means I think and they won't have to jump around. We're not even technically finding docblocks because that requires a proper parser. Really what we are looking for is "something that kind of looks like it could be a doc block"

Also, we're only using them once so we're not even getting any modularity benefits from extracting them.